cmd/srcd/cmd/components.go

```diff
@@ -60,30 +60,56 @@ var componentsInstallCmd = &cobra.Command{
 	Use:   "install [component]",
 	Short: "Install source{d} component",
 	Args:  cobra.MinimumNArgs(1),
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmps, err := components.List(context.Background(), false)
+		if err != nil {
+			return fmt.Errorf("could not list images: %s", err)
+		}
+
 		for _, arg := range args {
-			ok, err := components.IsInstalled(context.Background(), arg)
-			if err != nil {
-				if err == components.ErrNotSrcd {
-					log.Printf("can't install %s, docker image from unknown organization", arg)
-				} else {
-					log.Printf("could not check if %s is installed: %v", arg, err)
+			var c *components.Component
+			for _, cmp := range cmps {
+				// We allow to match by container name or by image name
+				if arg == cmp.Name || arg == cmp.Image || arg == cmp.ImageWithVersion() {
+					c = &cmp
+					break
 				}
-				os.Exit(1)
 			}
 
-			if ok {
-				log.Printf("%q is already installed", arg)
+			if c == nil {
+				names := make([]string, len(cmps))
+				for i, cmp := range cmps {
+					names[i] = cmp.Image
+				}
+
+				return fmt.Errorf("%s is not valid. Component must be one of [%s]", arg, strings.Join(names, ", "))
+			}
+
+			_, err = c.RetrieveVersion()
```

Comment:
this function would update version of the component right?
So if I do `srd components install srcd/some:0.1.0` it will actually install `srcd/some:0.1.298`?
It's unexpected.

Reply:
This is here to mimic the behaviour of `srcd init`. It's true that it will be confusing if the user includes the version in the argument, so I removed it in d85bdd7.
Now you won't be able to install `some:0.1.0`, it will not be a valid name.